packages/grid/src/Grid.tsx

```diff
@@ -1323,14 +1323,34 @@ class Grid extends PureComponent<GridProps, GridState> {
    * @param viewState New state properties to set.
    * @param forceUpdate Whether to force an update.
    */
-  setViewState(viewState: Partial<GridState>, forceUpdate = false): void {
+
+  setViewState(
+    viewState: Partial<GridState>,
+    forceUpdate = false,
+    eventType: WheelEvent | GridKeyboardEvent | null = null
+  ): void {
     if (!this.metrics) throw new Error('metrics not set');
 
     const { isStickyBottom, isStickyRight } = this.props;
     const { top, left } = viewState;
     const { lastTop, lastLeft } = this.metrics;
+    let isUserInputDown = false;
+
+    if (eventType instanceof WheelEvent) {
+      isUserInputDown = eventType.deltaY > 0;
+    } else if (
+      eventType instanceof KeyboardEvent ||
+      (eventType && 'nativeEvent' in eventType) // used to catch the case that a synthetic react keyboard event is passed
+    ) {
+      isUserInputDown =
+        eventType.key === 'ArrowDown' ||
+        eventType.key === 'End' ||
+        eventType.key === 'PageDown';
+    }
     if (top != null) {
-      this.setState({ isStuckToBottom: isStickyBottom && top >= lastTop });
+      this.setState({
+        isStuckToBottom: isStickyBottom && top >= lastTop && isUserInputDown,
+      });
     }
     if (left != null) {
       this.setState({ isStuckToRight: isStickyRight && left >= lastLeft });
```

Comment:
We should be mirroring the same logic for stuck to right. We don't have tables that grow by columns right now, but with pivot tables (coming at some point) it could happen.

Reply:
for checking if user input is right I would only be checking for the right arrow key right?